src/hooks/useCalculator.ts

```diff
@@ -0,0 +1,47 @@
+import { useCallback, useRef, useState } from "react";
+import { Calculator, OperatorSymbols } from "../calculator/calculator";
+
+interface useCalculatorOutput {
+  result: string;
+  pressNumber: (numberText: string) => void;
+  calculate: () => void;
+  pressOperator: (operatorSymbol: OperatorSymbols) => void;
+  allClear: () => void;
+}
+
+export default function useCalculator(): useCalculatorOutput {
+  const { current: calculator } = useRef<Calculator>(new Calculator());
+
+  const [result, setResult] = useState("0");
+
+  const pressNumber = useCallback(function pressNumber(numberText: string) {
+    calculator.pressNumber(numberText);
+    setResult(calculator.getDisplayValue());
+  }, []);
+
+  const calculate = useCallback(function calculate() {
+    calculator.calculate();
+    setResult(calculator.getDisplayValue());
+  }, []);
+
+  const pressOperator = useCallback(function pressOperator(
+    operatorSymbol: "+" | "-" | "*" | "/"
+  ) {
+    calculator.pressOperator(operatorSymbol);
+    setResult(calculator.getDisplayValue());
+  },
+  []);
+
+  const allClear = useCallback(function allClear() {
+    calculator.allClear();
+    setResult(calculator.getDisplayValue());
+  }, []);
```

Comment:
잘 보면 Calculator의 결과만 result에 반환하는 형태이고 함수도 중복으로 만드는 것 같네요.
중복코드를 많이 생성하는 것 처럼 보여서 조금 고민이네요.

```suggestion
  const { current: calculator } = useRef<Calculator>(new Calculator());

  const [result, setResult] = useState("0");

  useEffect(() => {
    setResult(calculator.displayValue);
  }, [calculator.displayValue]);
```

아예 displayValue를 getter로 만들어서 사용하면 이렇게 표현할 수 있지 않을까요 🤔